### PR TITLE
Per file auto format

### PR DESCRIPTION
**Pull request by harshad1**
This PR adds per-file auto format

- [x] Auto format defaults to true
- [x] Plain text files behave just like markdown
- [x] Fix behavior of auto-renumber on line move etc

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/format/AutoFormatter.java [hunk 1]
```diff
@@ -26,52 +26,35 @@
     private final PrefixPatterns _prefixPatterns;
     private final char _indentCharacter;
 
-    private CharSequence _source;
-    private int _start;
-    private int _end;
-    private Spanned _dest;
-    private int _dstart;
-    private int _dend;
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/AutoFormatter.java:34]: Removed unnecessary member vars

File: app/src/main/java/net/gsantner/markor/format/ListHandler.java [hunk 2]
```diff
@@ -17,16 +17,14 @@
 import net.gsantner.opoc.util.StringUtils;
 
 public class ListHandler implements TextWatcher {
-    private final boolean _reorderEnabled;
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/ListHandler.java:20]: Always enable reorder

File: app/src/main/java/net/gsantner/markor/format/TextFormat.java [hunk 3]
```diff
@@ -87,34 +92,38 @@ public static TextFormat getFormat(int formatId, Activity activity, Document doc
                 format.setConverter(CONVERTER_PLAINTEXT);
                 format.setHighlighter(new PlaintextHighlighter(hlEditor, document));
                 format.setTextActions(new PlaintextTextActions(activity, document));
+                format.setAutoFormatInputFilter(new MarkdownAutoFormat()); // Using the markdown syntax for plain text
+                format.setAutoFormatTextWatcher(new ListHandler(MarkdownAutoFormat.getPrefixPatterns()));
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/TextFormat.java:96]: Move selection of auto format from highlighter to here

File: app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java [hunk 4]
```diff
@@ -242,26 +242,21 @@ public void moveLineBy1(final boolean up) {
...
+            text.replace(Math.min(linesStart, altStart), Math.max(altEnd, linesEnd), newPair);
 
-                final String newPair = String.format("%s\n%s", up ? lines : altLine, up ? altLine : lines);
-                text.replace(Math.min(linesStart, altStart), Math.max(altEnd, linesEnd), newPair);
+            selStart[0] += up ? -1 : 1;
+            selEnd[0] += up ? -1 : 1;
 
-                selStart[0] += up ? -1 : 1;
-                selEnd[0] += up ? -1 : 1;
+            _hlEditor.withAutoFormatDisabled(() -> {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java:254]: Added `withAutoFormatDisabled` to simplify calling code for which auto formatters should not be active

File: app/src/main/java/net/gsantner/markor/format/TextFormat.java [hunk 5]
```diff
@@ -87,34 +92,38 @@ public static TextFormat getFormat(int formatId, Activity activity, Document doc
...
                 format.setHighlighter(new TodoTxtHighlighter(hlEditor, document));
                 format.setTextActions(new TodoTxtTextActions(activity, document));
+                format.setAutoFormatInputFilter(new TodoTxtAutoFormat());
                 break;
             }
-
             case FORMAT_KEYVALUE: {
                 format.setConverter(CONVERTER_KEYVALUE);
                 format.setHighlighter(new KeyValueHighlighter(hlEditor, document));
                 format.setTextActions(new PlaintextTextActions(activity, document));
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/format/TextFormat.java:109]: if nothing new is set, like here - the old autoformat is kept. thus if you had todotxt before and to K-V, it uses todo.txt autoformat there.   
correct?

**Comment by harshad1** [comment]
@gsantner @fgtham @fredericjacob Except for cleanups, I think this is mostly working.

Testing, comments etc would be appreciated

File: app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java [hunk 6]
```diff
@@ -128,14 +129,6 @@ public boolean runAction(String action) {
...
                 return true;
             }
-            case ACTION_MOVE_UP: {
-                moveLineBy1(true);
-                return true;
-            }
-            case ACTION_MOVE_DOWN: {
-                moveLineBy1(false);
-                return true;
-            }
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java:138]: Moved into TextActions. Eventually, I would like to get rid of `CommonTextActions`. Imo it is redundant with the `TextActions` base class

File: app/src/main/java/net/gsantner/markor/format/plaintext/PlaintextTextActions.java [hunk 7]
```diff
@@ -78,51 +79,30 @@ int getFormatActionsKey() {
...
+                case R.string.tmaid_common_indent:
+                    return runCommonTextAction(R.string.tmaid_common_set_indent_size);
+                case R.string.tmaid_common_open_link_browser:
+                    return runCommonTextAction(R.string.tmaid_common_search_in_content_of_current_file);
+                case R.string.tmaid_common_special_key:
+                    return runCommonTextAction(R.string.tmaid_common_jump_to_bottom);
+                case R.string.tmaid_common_time:
+                    return runCommonTextAction(R.string.tmaid_common_time_insert_timestamp);
+                default:
+                    return runAction(_action, true, null);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/plaintext/PlaintextTextActions.java:98]: Again, much nicer :)

Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/format/plaintext/PlaintextTextActions.java:102]: preferably add { braces }

**Comment by gsantner** [comment]
haven't installed it but changes seem fine to me

**Comment by harshad1** [comment]
@gsantner @fgtham @fredericjacob This is ready, imo

**Review by gsantner — APPROVED** [review]

**Pull request merged**